lib/src/upstream.rs

```diff
@@ -25,15 +26,17 @@ impl Connector {
     fn new(backend: &Backend) -> Self {
         Self {
             backend_uri: backend.uri.clone(),
-            https: HttpsConnector::new(),
+            https: HttpsConnector::with_native_roots(),
         }
     }
 }
 
+type BoxError = Box<dyn std::error::Error + Send + Sync>;
+
 impl hyper::service::Service<Uri> for Connector {
     type Response = MaybeHttpsStream<TcpStream>;
     type Error = Box<dyn std::error::Error + Send + Sync>;
```

Comment:
tiny nit, could we use the `BoxError` alias here as well, just to be consistent? stop me if there's some `rustc` wrinkle preventing that though :innocent: